plugin/storage/es/factory.go

```diff
@@ -317,9 +307,39 @@ func (f *Factory) CreateSamplingStore(maxBuckets int) (samplingstore.Store, erro
 		Lookback:               f.primaryConfig.AdaptiveSamplingLookback,
 		MaxDocCount:            f.primaryConfig.MaxDocCount,
 	})
+
+	if f.primaryConfig.CreateIndexTemplates && !f.primaryConfig.UseILM {
+		mappingBuilder := mappingBuilderFromConfig(f.primaryConfig)
+		samplingMapping, err := mappingBuilder.GetSamplingMappings()
+		if err != nil {
+			return nil, err
+		}
+		primaryClient := f.getPrimaryClient()
+		if primaryClient == nil {
+			return nil, errors.New("primary client is nil")
```

Comment:
does this even happen? You already called getPrimaryClient in L302

Reply:
CreateSamplingStore(), as all other CreateXYZ methods, are supposed to be called after the call to `factory.Initialize()`, which sets the primary client. So this null check is not needed.

Reply:
0e90660db25c69d63468aafab357b974cbf92d9e